opensearch/src/main/java/org/opensearch/sql/opensearch/request/OpenSearchQueryRequest.java

```diff
@@ -45,9 +59,19 @@ public class OpenSearchQueryRequest implements OpenSearchRequest {
   /** List of includes expected in the response. */
   @EqualsAndHashCode.Exclude @ToString.Exclude private final List<String> includes;
 
+  @EqualsAndHashCode.Exclude private boolean needClean = true;
+
   /** Indicate the search already done. */
   private boolean searchDone = false;
 
+  private String pitId;
```

Comment:
Add a new PIT request class instead? I think all the conditional check added make this class hard to maintain.

Reply:
I've captured this improvement as part of this issue -https://github.com/opensearch-project/sql/issues/2945